### PR TITLE
ER_623: UNI•Login improvements

### PR DESCRIPTION
**Pull request by rimi-itk**
https://jira.itkdev.dk/browse/ER-623

Adds whitelisting of institutions assigned to users signing in via UNI•Login.

### CODE REVIEW

File: ding_unilogin/lib/DingUniloginUser.class.inc [hunk 1]
```diff
@@ -121,24 +121,81 @@ class DingUniloginUser {
...
-    foreach ($institutions as $institution_id) {
-      $institution = $this->getInstitutionService()->getInstitution($institution_id);
-      if (isset($institution->kommunenr)) {
-        $municipality_ids[] = $institution->kommunenr;
+      if (NULL === $institutionIds) {
+        $institutionIds = $this->getInstitutionsIds();
+      }
+      foreach ($institutionIds as $institutionId) {
+        $institution = $this->getInstitutionService()
+          ->getInstitution($institutionId);
```
Comment (cableman) [inline ding_unilogin/lib/DingUniloginUser.class.inc]: Why this on a new-line!

File: ding_unilogin/ding_unilogin.admin.inc [hunk 2]
```diff
@@ -90,3 +90,124 @@ function ding_unilogin_admin_settings_form($form, &$form_state) {
...
+}
+
+/**
+ * Submit handler for UNI•Login institutions form.
+ */
+function ding_unilogin_admin_institutions_form_submit(array $form, array &$form_state) {
+  $values = $form_state['values'];
+
+  // Get institutions with id.
+  $institutions = array_filter($values['ding_unilogin_institutions'], static function (array $institution) {
```
[the remark this answers is not in the document]
Comment (rimi-itk) [reply ding_unilogin/ding_unilogin.admin.inc]: Please see https://github.com/eReolen/ereolengo/pull/4/files#diff-62071cd5b4c8bb313be26a4a32f5688aR123

File: ding_unilogin/ding_unilogin.admin.inc [hunk 3]
```diff
@@ -90,3 +90,124 @@ function ding_unilogin_admin_settings_form($form, &$form_state) {
...
+
+  $form['ding_unilogin_institutions'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('Institutions'),
+    '#tree' => TRUE,
+    '#description' => t('Clear the institution id to remove a row.'),
+  );
+
+  // Add empty row.
+  $institutions['__new__'] = array(
```
[the remark this answers is not in the document]
Comment (rimi-itk) [reply ding_unilogin/ding_unilogin.admin.inc]: I know, but I'm lazy and hopefully this value will soon be updated automatically from a webservice (https://viden.stil.dk/pages/viewpage.action?pageId=2360660) making this form mostly obsolete.

File: ding_unilogin/lib/DingUniloginUser.class.inc (hunk 1 above)
Comment (rimi-itk) [reply ding_unilogin/lib/DingUniloginUser.class.inc]: Don't the Drupal coding standards call for lines to not contain more than 80 characters?

File: ding_unilogin/ding_unilogin.module [hunk 4]
```diff
@@ -444,7 +457,11 @@ function _ding_unilogin_login($username) {
   $account = _ding_user_create_account($user->getAuthName(), $auth_res);
 
   // Make a note that this is a UNI•Login account in data by saving once more.
-  user_save($account, array('data' => array('unilogin' => TRUE)));
+  $unilogin_user = user_save($account, array(
+    'data' => array(
+      'unilogin' => $unilogin_data,
```
[the remark this answers is not in the document]
Comment (rimi-itk) [reply ding_unilogin/ding_unilogin.module]: Moved this data into `$_SESSION`.

**Pull request merged**